src/net/echo.zig

```diff
@@ -21,28 +15,27 @@ const HEADER_LENGTH: usize = 4;
 /// comes from the snapshot, and ip echo is only used to validate it.
 pub fn getShredAndIPFromEchoServer(
     logger: Logger,
-    allocator: std.mem.Allocator,
-    socket_addresses: []SocketAddr,
+    socket_addresses: []const SocketAddr,
 ) !struct { shred_version: ?u16, ip: ?IpAddr } {
     var my_ip: ?IpAddr = null;
     var my_shred_version: ?u16 = null;
 
     for (socket_addresses) |socket_addr| {
-        if (requestIpEcho(allocator, socket_addr.toAddress(), .{})) |response| {
-            if (my_ip == null)
-                my_ip = response.address;
-
-            if (response.shred_version) |shred_version| {
-                logger.info()
-                    .field("shred_version", shred_version.value)
-                    .field("ip", socket_addr.toString().constSlice())
-                    .log("ip echo response");
-                my_shred_version = shred_version.value;
-            }
+        const response = requestIpEcho(socket_addr.toAddress(), .{}) catch continue;
+        my_ip = my_ip orelse response.address;
+        if (my_ip == null)
+            my_ip = response.address;
```

Comment:
isnt this if redundant?

Reply:
Missed that whoops

Reply:
e74ef3fb1ccbd0b50469af7798d3ebc02110e31f